packages/opentelemetry-core/src/trace/instrumentation/BasePlugin.ts

```diff
@@ -28,7 +28,7 @@ import * as path from 'path';
 /** This class represent the base to patch plugin. */
 export abstract class BasePlugin<T> implements Plugin<T> {
   supportedVersions?: string[];
-  readonly moduleName?: string; // required for internalFilesExports
+  readonly moduleName!: string; // required for internalFilesExports
```

Comment:
If you make this abstract, then you don't need to assert it with ! and the type checker will ensure it is used correctly

Reply:
Didn't know that, thanks for the tip !